File: src/main/scala/org/openapitools/client/api/RecordRevisionList.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import RecordRevisionList._

case class RecordRevisionList (
  metadata: Option[Metadata],
pagination: Option[Pagination],
data: Option[List[RecordRevision]])

object RecordRevisionList {
  import DateTimeCodecs._

  implicit val RecordRevisionListCodecJson: CodecJson[RecordRevisionList] = CodecJson.derive[RecordRevisionList]
  implicit val RecordRevisionListDecoder: EntityDecoder[RecordRevisionList] = jsonOf[RecordRevisionList]
  implicit val RecordRevisionListEncoder: EntityEncoder[RecordRevisionList] = jsonEncoderOf[RecordRevisionList]
}

File: API/Client/RecordRevisionList.js
goog.provide('API.Client.RecordRevisionList');

/**
 * @record
 */
API.Client.RecordRevisionList = function() {}

/**
 * @type {!API.Client.Metadata}
 * @export
 */
API.Client.RecordRevisionList.prototype.metadata;

/**
 * @type {!API.Client.Pagination}
 * @export
 */
API.Client.RecordRevisionList.prototype.pagination;

/**
 * @type {!Array<!API.Client.RecordRevision>}
 * @export
 */
API.Client.RecordRevisionList.prototype.data;

